src/components/data/data-utils.js

```diff
@@ -311,36 +311,32 @@ export const EntityUploader = ({ onSuccess, onDismiss, namespace, name, entityTy
   const [showInvalidEntryMethodWarning, setShowInvalidEntryMethodWarning] = useState(false)
   const [uploading, setUploading] = useState(false)
   const [deleteEmptyValues, setDeleteEmptyValues] = useState(false)
+  const [recordType, setRecordType] = useState(undefined)
+  const [recordAlreadyExists, setRecordAlreadyExists] = useState(false)
 
   const doUpload = async () => {
     setUploading(true)
     try {
-      const workspace = Ajax().Workspaces.workspace(namespace, name)
-      if (useFireCloudDataModel) {
-        await workspace.importEntitiesFile(file, { deleteEmptyValues })
-      } else {
-        const filesize = file?.size || Number.MAX_SAFE_INTEGER
-        if (filesize < 524288) { // 512k
-          await workspace.importFlexibleEntitiesFileSynchronous(file, { deleteEmptyValues })
-        } else {
-          const { jobId } = await workspace.importFlexibleEntitiesFileAsync(file, { deleteEmptyValues })
-          asyncImportJobStore.update(Utils.append({ targetWorkspace: { namespace, name }, jobId }))
-          notifyDataImportProgress(jobId)
-        }
-      }
+      await dataProvider.doUpload({ workspaceId, recordType, file, useFireCloudDataModel, deleteEmptyValues, namespace, name })
       onSuccess()
       Ajax().Metrics.captureEvent(Events.workspaceDataUpload, {
         workspaceNamespace: namespace, workspaceName: name
-      })
+      }) //do we do this for WDS?
```

Comment:
Is this an artifact? Or should we address?

Reply:
Should be addressed - will ultimately be covered in https://broadworkbench.atlassian.net/browse/AJ-656, but just need to make sure we're not messing up metrics in the meantime.

Reply:
If you wouldn't mind, add a `TODO` comment here to make it easier to find when we tackle AJ-656, e.g.:

https://github.com/DataBiosphere/terra-ui/blob/359dfc4fb4a6f5c7e4474e7a0340e70f87685bbc/src/pages/workspaces/workspace/Data.js#L339